src/Entity/ShippingMethod.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Shipping Slot plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusShippingSlotPlugin\Entity;
+
+use Sylius\Component\Core\Model\ShippingMethod as SyliusShippingMethod;
+
+class ShippingMethod extends SyliusShippingMethod implements ShippingMethodInterface
```

Comment:
Can we have a Trait instead? Since an object can't extend more than one abstract class.

And you can use annotations in the trait, instead of an XML file for Doctrine.